workflow/rocoto/gfs_cycled_xml.py

```diff
@@ -24,6 +24,17 @@ def get_cycledefs(self):
         sdate_str = sdate.strftime("%Y%m%d%H%M")
         strings.append(f'\t<cycledef group="gdas">{sdate_str} {edate_str} {interval_str}</cycledef>')
 
+        if self._app_config.do_jedisnowda:
+            sdate_snocvr = self._base['SDATE']
+            edate_snocvr = self._base['EDATE']
+            interval_snocvr = to_timedelta('24H')
+            sdate_snocvr = sdate_snocvr + interval_snocvr
+            sdate_snocvr_str = sdate_snocvr.replace(hour=0, minute=0, second=0).strftime("%Y%m%d%H%M")
+            edate_snocvr_str = edate_snocvr.strftime("%Y%m%d%H%M")
+            interval_str = timedelta_to_HMS(interval_snocvr)
+            if sdate_snocvr.date() <= edate_snocvr.date():
+                strings.append(f'\t<cycledef group="gdas_prep_snocvr">{sdate_snocvr_str} {edate_snocvr_str} {interval_str}</cycledef>')
+
```

Comment:
Actually, I don't think that works for the edge case of a warm start at 00Z.  Instead, I think we need to find out if the first full cycle is after 06Z and the last full cycle is on the same day.  So let's change the if block a bit instead.  This renders identical rocotostat output for me.

```suggestion        if self._app_config.do_jedisnowda:
        if self._app_config.do_jedisnowda:
            sdate = self._base['SDATE']
            edate_snocvr = self._base['EDATE'].replace(hour=0, minute=0, second=0)
            interval_snocvr = to_timedelta('24H')
            assim_freq = to_timedelta("{self._base['assim_freq']}H")

            is_warm_start = self._base.get('EXP_WARM_START', False)
            first_full_cycle = sdate if is_warm_start else sdate + assim_freq
            sdate_snocvr = (first_full_cycle if first_full_cycle.hour == 0 else
                            (first_full_cycle + interval_snocvr).replace(hour=0, minute=0, second=0))

            sdate_snocvr_str = sdate_snocvr.strftime("%Y%m%d%H%M")
            edate_snocvr_str = edate_snocvr.strftime("%Y%m%d%H%M")
            interval_str = timedelta_to_HMS(interval_snocvr)

            if sdate_snocvr <= edate_snocvr:
                strings.append(f'\t<cycledef group="gdas_prep_snocvr">{sdate_snocvr_str} {edate_snocvr_str} {interval_str}</cycledef>')
```

Reply:
Thanks @DavidHuber-NOAA for your great efforts. I made the corresponding changes and committed the changes. The `rocotostat` produced the same workflow for my test cases as previously, but I don't know how to examine the warm start case.

Reply:
I think that's OK for now since it isn't a case we are supporting yet.